### DISCUSSION TITLE
Koka and Category Theory

### DISCUSSION THREAD

**Discussion by chtenb** [koka-lang/koka#469] [General]
I'm starting to learn a little bit about category theory, and I know that the category `Set` can be used to model total functional programming, and `Hask` can be used to model pure functional programming. Is there a category that models effectful functional programming, i.e. `Koka`?

> The precise effect typing gives Koka rock-solid semantics backed by well-studied category theory

This statement on the homepage suggested to me the answer is yes :) What would it look like?

**Comment by TimWhiting** [comment]
The Algebraic Effects feature was introduced into languages based on more Mathematical / Categorial origins. I think this is the canonical paper: [A Logic for Algebraic Effects](https://ieeexplore.ieee.org/abstract/document/4557905) by Plotkin & Pretnar, followed up by an implementation of handlers in Pretnar's dissertation [The Logic And Handling of Algebraic Effects](https://era.ed.ac.uk/bitstream/handle/1842/4611/Pretnar2010.pdf)

Here is a good resource for seeing the history of publications in algebraic effects. https://github.com/yallop/effects-bibliography
